scripts/Test-Signatures.ps1

```diff
@@ -0,0 +1,209 @@
+    # "PackageId" = relative paths
+    "Microsoft.TestPlatform" = @( `
+        ".\tools\net451\Common7\IDE\Extensions\TestPlatform\Newtonsoft.Json.dll" `
+    )
+
+    "Microsoft.TestPlatform.CLI" = @( `
+        ".\contentFiles\any\netcoreapp2.1\TestHost\Newtonsoft.Json.dll", `
+        ".\contentFiles\any\netcoreapp2.1\Newtonsoft.Json.dll"
+    )
+    
+    "Microsoft.TestPlatform.Portable" = @( `
+        ".\tools\net451\Newtonsoft.Json.dll", `
+        ".\tools\netcoreapp2.1\TestHost\Newtonsoft.Json.dll", `
+        ".\tools\netcoreapp2.1\Newtonsoft.Json.dll"
+    )
+}
+
+
+$global:Depth = 0;
+
+function Get-RelativePath($path, $relativeTo) {
+    Push-Location
+    Set-Location $relativeTo
+    $path = Resolve-Path $path -Relative
+    Pop-Location
+    
+    return $path
+}
+
+function Push-Depth {
+    $global:Depth = $global:Depth + 1
+}
+
+function Pop-Depth {
+    $global:Depth = $global:Depth - 1
+
+    if($global:Depth -lt 0) {
+        $global:Depth = 0;
+    }
+}
+
+function Clear-Line($message = "") {
+    if(-not $CIBuild) {
+        $x = [System.Console]::get_CursorLeft()
+        Write-Host "`r$(' ' * $x)$message`r" -NoNewline
+    }
+}
+
+function Update-Operation($operation, $message, [switch]$NewLine, $suppress=$false) {
+    if($suppress) {
+        return;
+    }
+    
+    Clear-Line;
+    Write-Host "$('  ' * $Depth)[$operation] " -ForegroundColor Yellow -NoNewline
+    Write-Host $message -NoNewline
+
+    if ($newline -or $CIBuild) {
+        Write-Host;
+    }
+}
+
+function Submit-Operation($operation, $message, [switch]$NoNewLine, $suppress=$false) {
+    if($suppress) {
+        return;
+    }
+    
+    Clear-Line;
+    Write-Host "$('  ' * $Depth)[$operation] " -ForegroundColor Green -NoNewline
+    Write-Host $message -NoNewline
+
+    if (-not $NoNewLine -or $CIBuild) {
+        Write-Host;
+    }
+}
+
+function Undo-Operation($operation, $message, [switch]$NoNewLine, $suppress=$false) {
+    if($suppress) {
+        return;
+    }
+    
+    Clear-Line;
+    Write-Host "$('  ' * $Depth)[$operation] " -ForegroundColor Red -NoNewline
+    Write-Host $message -NoNewline
+
+    if (-not $NoNewLine -or $CIBuild) {
+        Write-Host;
+    }
+}
+
+function Expand-NugetPackages {
+    $foldersToProcess = [System.Collections.Generic.HashSet[string]]@()
+
+    $nupgks = Get-ChildItem *.nupkg
+
+    Push-Depth
+    $nupgks | ForEach-Object {
+        $package = $_.FullName
+        $expand = Join-Path -Path ([System.IO.Path]::GetDirectoryName($package)) -ChildPath ([System.IO.Path]::GetFileNameWithoutExtension($package))
+        ((Test-Path -Path $expand) -and (Remove-Item $expand -Force -Recurse)) | Out-Null
+
+        Update-Operation -operation "extracting" -message "$([System.IO.Path]::GetFileName($package))"
+        [System.IO.Compression.ZipFile]::ExtractToDirectory($package, $expand)
+        Submit-Operation -operation "done" -message "$([System.IO.Path]::GetFileName($package))" -suppress $CIBuild

```

Comment:
This won't work properly if something fails. You would need to wrap that into finally, but that would suck.